lib/ansible/plugins/strategy/__init__.py

```diff
@@ -839,8 +839,12 @@ def run_handlers(self, iterator, play_context):
             for handler in handler_block.block:
                 if handler.notified_hosts:
                     handler_vars = self._variable_manager.get_vars(play=iterator._play, task=handler)
-                    handler_name = handler.get_name()
-                    result = self._do_handler_run(handler, handler_name, iterator=iterator, play_context=play_context)
+                    if not ((handler.action == 'include' and handler.evaluate_tags([], play_context.skip_tags, all_vars=handler_vars))
+                            or handler.evaluate_tags(play_context.only_tags, play_context.skip_tags, all_vars=handler_vars)):
+                        continue
```

Comment:
if im reading this correctly, we check tags for handlers, unless its an include, then we only check skip_tags?

Reply:
Yes, currently we check for the "include" special case for regular tasks too - https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/block.py#L376 - e174247734f28a26cd4ca9d552fd79baa3b1577f. Would that not apply to handler tasks?